File: Python/2095.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteMiddle(self, head):
        lst = []
        n = 0
        a = head
        while(a != None):
            n += 1
            a = a.next
        if n == 1:
            return None
        mid = n//2
        a = head
        for i in range(n):
            if i+1 == mid:
                a.next = a.next.next
                break
            a = a.next
        return head

File: Java/2095.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        int n = 0, mid = 0;
        ListNode a = new ListNode();
        a = head;
        while(a != null){
            n++;
            a = a.next;
        }
        if(n==1) return null;
        mid = n/2;
        a = head;
        for(int i = 0; i < mid-1; i++) a = a.next;
        a.next = a.next.next;
        return head;
    }
}